File: bls_jztc/demo/api/backend/v1/file.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
	"github.com/gogf/gf/v2/net/ghttp"
)

// FileUploadReq 文件上传请求
type FileUploadReq struct {
	g.Meta    `path:"/file/upload" method:"post" mime:"multipart/form-data" tags:"文件管理" summary:"上传文件" security:"Bearer" description:"上传文件到阿里云OSS，需要管理员权限"`
	File      *ghttp.UploadFile `type:"file" v:"required#请选择需要上传的文件" dc:"上传的文件"`
	Directory string            `dc:"存储目录，选填，默认为配置中的目录"`
	IsPublic  bool              `dc:"是否公开访问，选填，默认为配置中的设置"`
}

// FileUploadRes 文件上传响应
type FileUploadRes struct {
	g.Meta     `mime:"application/json" example:"json"`
	Id         int    `json:"id" dc:"文件ID"`
	Name       string `json:"name" dc:"文件名"`
	Url        string `json:"url" dc:"文件访问链接"`
	Size       int64  `json:"size" dc:"文件大小(字节)"`
	Type       string `json:"type" dc:"文件类型"`
	Extension  string `json:"extension" dc:"文件扩展名"`
	IsPublic   bool   `json:"isPublic" dc:"是否公开访问"`
	UploadTime string `json:"uploadTime" dc:"上传时间"`
}

// FileListReq 文件列表请求
type FileListReq struct {
	g.Meta   `path:"/file/list" method:"get" tags:"文件管理" summary:"获取文件列表" security:"Bearer" description:"获取文件列表，支持分页和搜索，需要管理员权限"`
	Page     int    `d:"1" v:"min:1#页码最小为1" dc:"页码，默认1"`
	PageSize int    `d:"10" v:"min:1#每页大小最小为1" dc:"每页数量，默认10"`
	Keyword  string `dc:"搜索关键词，支持文件名搜索"`
	Type     string `dc:"文件类型筛选，例如：image、document等"`
	IsPublic *bool  `dc:"是否公开访问筛选，可选值：true/false"`
}

// FileListRes 文件列表响应
type FileListRes struct {
	g.Meta `mime:"application/json" example:"json"`
	List   []FileInfo `json:"list" dc:"文件列表"`
	Total  int        `json:"total" dc:"总数量"`
	Page   int        `json:"page" dc:"当前页码"`
}

// FileInfo 文件信息结构
type FileInfo struct {
	Id          int    `json:"id" dc:"文件ID"`
	Name        string `json:"name" dc:"文件名"`
	Url         string `json:"url" dc:"文件访问链接"`
	Path        string `json:"path" dc:"文件存储路径"`
	Size        int64  `json:"size" dc:"文件大小(字节)"`
	SizeFormat  string `json:"sizeFormat" dc:"文件大小格式化"`
	Type        string `json:"type" dc:"文件类型"`
	ContentType string `json:"contentType" dc:"内容类型"`
	Extension   string `json:"extension" dc:"文件扩展名"`
	IsPublic    bool   `json:"isPublic" dc:"是否公开访问"`
	UserId      int    `json:"userId" dc:"上传用户ID"`
	Username    string `json:"username" dc:"上传用户名"`
	CreatedAt   string `json:"createdAt" dc:"创建时间"`
}

// FileDeleteReq 删除文件请求
type FileDeleteReq struct {
	g.Meta `path:"/file/delete" method:"delete" tags:"文件管理" summary:"删除文件" security:"Bearer" description:"从阿里云OSS删除文件，需要管理员权限"`
	Id     int `v:"required#文件ID不能为空" dc:"文件ID"`
}

// FileDeleteRes 删除文件响应
type FileDeleteRes struct {
	g.Meta  `mime:"application/json" example:"json"`
	Success bool   `json:"success" dc:"是否成功"`
	Message string `json:"message" dc:"提示信息"`
}

// FileBatchDeleteReq 批量删除文件请求
type FileBatchDeleteReq struct {
	g.Meta `path:"/file/batch-delete" method:"delete" tags:"文件管理" summary:"批量删除文件" security:"Bearer" description:"从阿里云OSS批量删除文件，需要管理员权限"`
	Ids    []int `v:"required#文件ID列表不能为空" dc:"文件ID列表"`
}

// FileBatchDeleteRes 批量删除文件响应
type FileBatchDeleteRes struct {
	g.Meta  `mime:"application/json" example:"json"`
	Success bool   `json:"success" dc:"是否成功"`
	Message string `json:"message" dc:"提示信息"`
	Count   int    `json:"count" dc:"成功删除数量"`
}

// FileDetailReq 文件详情请求
type FileDetailReq struct {
	g.Meta `path:"/file/detail" method:"get" tags:"文件管理" summary:"获取文件详情" security:"Bearer" description:"获取文件详情信息，需要管理员权限"`
	Id     int `v:"required#文件ID不能为空" dc:"文件ID"`
}

// FileDetailRes 文件详情响应
type FileDetailRes struct {
	g.Meta `mime:"application/json" example:"json"`
	FileInfo
}

// FileUpdatePublicReq 更新文件公开状态请求
type FileUpdatePublicReq struct {
	g.Meta   `path:"/file/update-public" method:"put" tags:"文件管理" summary:"更新文件公开状态" security:"Bearer" description:"更新文件的公开访问状态，需要管理员权限"`
	Id       int  `v:"required#文件ID不能为空" dc:"文件ID"`
	IsPublic bool `dc:"是否公开访问"`
}

// FileUpdatePublicRes 更新文件公开状态响应
type FileUpdatePublicRes struct {
	g.Meta  `mime:"application/json" example:"json"`
	Success bool   `json:"success" dc:"是否成功"`
	Message string `json:"message" dc:"提示信息"`
	Url     string `json:"url" dc:"文件新的访问链接"`
}

File: bls_jztc/jztc/src/api/file.ts
import request from '@/utils/request';

// 通用响应结构
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// 文件信息接口
export interface FileInfo {
  id: number;
  name: string;
  path: string;
  url: string;
  size: number;
  sizeFormat: string;
  type: string;
  extension: string;
  contentType: string;
  isPublic: boolean;
  userId: number;
  username: string;
  createdAt: string;
}

// 文件列表请求参数
export interface FileListReq {
  Page?: number;
  PageSize?: number;
  Keyword?: string;
  Type?: string;
  IsPublic?: boolean;
}

// 文件列表响应
export interface FileListRes {
  code: number;
  message: string;
  data: {
    list: FileInfo[];
    total: number;
    page: number;
  };
}

// 文件上传请求参数
export interface FileUploadReq {
  File: File;
  Directory?: string;
  IsPublic?: boolean;
}

// 文件上传响应
export interface FileUploadRes {
  id: number;
  name: string;
  extension: string;
  size: number;
  type: string;
  url: string;
  isPublic: boolean;
  uploadTime: string;
}

// 文件更新公开状态请求参数
export interface FileUpdatePublicReq {
  Id: number;
  IsPublic: boolean;
}

// 文件更新公开状态响应
export interface FileUpdatePublicRes {
  success: boolean;
  message: string;
  url: string;
}

// 文件删除响应
export interface FileDeleteRes {
  success: boolean;
  message: string;
}

// 文件批量删除响应
export interface FileBatchDeleteRes {
  success: boolean;
  message: string;
  count: number;
}

/**
 * 获取文件列表
 * @param params 查询参数
 * @returns 文件列表
 */
export const getFileList = (params: FileListReq): Promise<FileListRes> => {
  return request.get('/file/list', { params });
};

/**
 * 获取文件详情
 * @param id 文件ID
 * @returns 文件详情
 */
export const getFileDetail = (id: number): Promise<ApiResponse<FileInfo>> => {
  return request.get('/file/detail', { params: { Id: id } });
};

/**
 * 上传文件
 * @param data 上传文件数据
 * @returns 上传结果
 */
export const uploadFile = (data: FormData): Promise<ApiResponse<FileUploadRes>> => {
  return request({
    url: '/file/upload',
    method: 'post',
    data,
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
};

/**
 * 更新文件公开状态
 * @param data 请求参数
 * @returns 更新结果
 */
export const updateFilePublic = (data: FileUpdatePublicReq): Promise<ApiResponse<FileUpdatePublicRes>> => {
  return request({
    url: '/file/update-public',
    method: 'put',
    data,
  });
};

/**
 * 删除文件
 * @param id 文件ID
 * @returns 删除结果
 */
export const deleteFile = (id: number): Promise<ApiResponse<FileDeleteRes>> => {
  return request({
    url: '/file/delete',
    method: 'delete',
    params: { Id: id },
  });
};

/**
 * 批量删除文件
 * @param ids 文件ID列表
 * @returns 批量删除结果
 */
export const batchDeleteFiles = (ids: number[]): Promise<ApiResponse<FileBatchDeleteRes>> => {
  return request({
    url: '/file/batch-delete',
    method: 'delete',
    params: { Ids: ids },
  });
}; 